bot/exts/utils/snekbox/_cog.py

```diff
@@ -298,86 +265,173 @@ async def format_output(self, output: str) -> Tuple[str, Optional[str]]:
         lines = output.count("\n")
 
         if lines > 0:
-            output = [f"{i:03d} | {line}" for i, line in enumerate(output.split('\n'), 1)]
-            output = output[:11]  # Limiting to only 11 lines
+            if line_nums:
+                output = [f"{i:03d} | {line}" for i, line in enumerate(output.split('\n'), 1)]
```

Comment:
There's an issue with this transformation being conditional. If `lines_nums` is False, `output` stays a string, and then `"\n".join(output)` a few lines later breaks it into one character per line.